### PR TITLE
[Ops][BugFix] Fix SiluAndMulClamp key mismatch in REGISTERED_ASCEND_OPS

### LINKED ISSUE #16500: [Bug]:REGISTERED_ASCEND_OPS 中 SiluAndMulWithClamp 的 key 不匹配，导致 Ascend 融合算子从未生效
### Your current environment

<details>
<summary>The output of `python collect_env.py`</summary>

```text
INFO 09-14 06:18:57 [__init__.py:44] Available plugins for group vllm.platform_plugins:
INFO 09-14 06:18:57 [__init__.py:46] - ascend -> vllm_ascend:register
INFO 09-14 06:18:57 [__init__.py:49] All plugins in this group will be loaded. Set `VLLM_PLUGINS` to control which plugins to load.
INFO 09-14 06:18:57 [__init__.py:237] Platform plugin ascend is activated
INFO 09-14 06:18:57 [platform.py:61] Breakable cudagraph is force disabled on Ascend because DeepSeek V4 PIECEWISE cudagraph is not supported yet.
Collecting environment information...
PyTorch version: 2.10.0+cpu
Is debug build: False

OS: Ubuntu 22.04.5 LTS (aarch64)
GCC version: (Ubuntu 11.4.0-1ubuntu1~22.04.3) 11.4.0
Clang version: 15.0.7
CMake version: version 4.4.0
Libc version: glibc-2.35

Python version: 3.12.13 (main, Jul  7 2026, 16:29:36) [GCC 11.4.0] (64-bit runtime)
Python platform: Linux-5.10.0-136.12.0.86.h1903.eulerosv2r12.aarch64-aarch64-with-glibc2.35

CPU:
Architecture:                         aarch64
CPU op-mode(s):                       64-bit
Byte Order:                           Little Endian
CPU(s):                               192
On-line CPU(s) list:                  0-191
Vendor ID:                            HiSilicon
BIOS Vendor ID:                       HiSilicon
Model name:                           Kunpeng-920
BIOS Model name:                      HUAWEI Kunpeng 920 5250
Model:                                0
Thread(s) per core:                   1
Core(s) per socket:                   48
Socket(s):                            4
Stepping:                             0x1
BogoMIPS:                             200.00
Flags:                                fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma dcpop asimddp asimdfhm ssbs
L1d cache:                            12 MiB (192 instances)
L1i cache:                            12 MiB (192 instances)
L2 cache:                             96 MiB (192 instances)
L3 cache:                             192 MiB (8 instances)
NUMA node(s):                         8
NUMA node0 CPU(s):                    0-23
NUMA node1 CPU(s):                    24-47
NUMA node2 CPU(s):                    48-71
NUMA node3 CPU(s):                    72-95
NUMA node4 CPU(s):                    96-119
NUMA node5 CPU(s):                    120-143
NUMA node6 CPU(s):                    144-167
NUMA node7 CPU(s):                    168-191
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; __user pointer sanitization
Vulnerability Spectre v2:             Not affected
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] numpy==1.26.4
[pip3] pyzmq==26.4.0
[pip3] torch==2.10.0+cpu
[pip3] torch_npu==2.10.0.post2
[pip3] torchaudio==2.10.0+cpu
[pip3] torchvision==0.25.0+cpu
[pip3] transformers==5.14.1
[conda] Could not collect
vLLM Version: 0.25.1
vLLM Ascend Version: 0.19.1rc2.dev1239+g2b1ac6124 (git sha: 2b1ac6124)

ENV Variables:
ASCEND_TOOLKIT_LATEST_HOME=/usr/local/Ascend/ascend-toolkit/latest
ATB_OPSRUNNER_KERNEL_CACHE_LOCAL_COUNT=1
ATB_STREAM_SYNC_EVERY_RUNNER_ENABLE=0
ASCEND_CUSTOM_PATH=/usr/local/Ascend/cann-9.0.1
ATB_STREAM_SYNC_EVERY_KERNEL_ENABLE=0
ATB_CXX_ABI=1
ATB_OPSRUNNER_KERNEL_CACHE_GLOABL_COUNT=5
ATB_HOME_PATH=/usr/local/Ascend/nnal/atb/latest/atb/cxx_abi_1
ASCEND_TOOLKIT_HOME=/usr/local/Ascend/cann-9.0.1
ATB_COMPARE_TILING_EVERY_KERNEL=0
ASCEND_OPP_PATH=/usr/local/Ascend/cann-9.0.1/opp
LD_LIBRARY_PATH=/usr/local/Ascend/cann-9.0.1/opp/vendors/gtsops/op_api/lib:/usr/local/Ascend/cann-9.0.1/acllib/lib64:/usr/local/Ascend/cann-9.0.1/tools/profiler/lib64:/usr/local/Ascend/cann-9.0.1/lib64:/usr/local/Ascend/cann-9.0.1/lib64/plugin/opskernel:/usr/local/Ascend/cann-9.0.1/lib64/plugin/nnengine:/usr/local/Ascend/cann-9.0.1/opp/built-in/op_impl/ai_core/tbe/op_tiling:/usr/local/Ascend/driver/lib64:/usr/local/Ascend/driver/lib64/common:/usr/local/Ascend/driver/lib64/driver:/usr/local/Ascend/nnal/atb/latest/atb/cxx_abi_1/lib:/usr/local/Ascend/nnal/atb/latest/atb/cxx_abi_1/examples:/usr/local/Ascend/nnal/atb/latest/atb/cxx_abi_1/tests/atbopstest:/usr/local/Ascend/ascend-toolkit/latest/tools/aml/lib64:/usr/local/Ascend/ascend-toolkit/latest/tools/aml/lib64/plugin:/usr/local/Ascend/ascend-toolkit/latest/lib64:/usr/local/Ascend/ascend-toolkit/latest/lib64/plugin/opskernel:/usr/local/Ascend/ascend-toolkit/latest/lib64/plugin/nnengine:/usr/local/Ascend/ascend-toolkit/latest/opp/built-in/op_impl/ai_core/tbe/op_tiling:/usr/local/python3.12.13/lib::/usr/local/lib:/usr/local/Ascend/cann-9.0.1/acllib/lib64/stub
ASCEND_AICPU_PATH=/usr/local/Ascend/cann-9.0.1
ATB_STREAM_SYNC_EVERY_OPERATION_ENABLE=0
ASCEND_HOME_PATH=/usr/local/Ascend/cann-9.0.1
ATB_MATMUL_SHUFFLE_K_ENABLE=1
ATB_WORKSPACE_MEM_ALLOC_ALG_TYPE=1
ATB_SHARE_MEMORY_NAME_SUFFIX=
TORCH_DEVICE_BACKEND_AUTOLOAD=1
TORCHINDUCTOR_CACHE_DIR=/tmp/torchinductor_root
VLLM_DISABLE_SHARED_EXPERTS_STREAM=1
PYTORCH_NVML_BASED_CUDA_CHECK=1
TORCHINDUCTOR_COMPILE_THREADS=1


NPU:
+------------------------------------------------------------------------------------------------+
| npu-smi 25.2.0                   Version: 25.2.0                                               |
+---------------------------+---------------+----------------------------------------------------+
| NPU   Name                | Health        | Power(W)    Temp(C)           Hugepages-Usage(page)|
| Chip                      | Bus-Id        | AICore(%)   Memory-Usage(MB)  HBM-Usage(MB)        |
+===========================+===============+====================================================+
| 0     910B3               | OK            | 96.1        49                0    / 0             |
| 0                         | 0000:C1:00.0  | 0           0    / 0          58669/ 65536         |
+===========================+===============+====================================================+
| 1     910B3               | OK            | 96.5        51                0    / 0             |
| 0                         | 0000:01:00.0  | 0           0    / 0          58669/ 65536         |
+===========================+===============+====================================================+
| 2     910B3               | OK            | 94.9        51                0    / 0             |
| 0                         | 0000:C2:00.0  | 0           0    / 0          58669/ 65536         |
+===========================+===============+====================================================+
| 3     910B3               | OK            | 99.3        52                0    / 0             |
| 0                         | 0000:02:00.0  | 0           0    / 0          58669/ 65536         |
+===========================+===============+====================================================+
| 4     910B3               | OK            | 93.7        49                0    / 0             |
| 0                         | 0000:81:00.0  | 0           0    / 0          58669/ 65536         |
+===========================+===============+====================================================+
| 5     910B3               | OK            | 97.6        50                0    / 0             |
| 0                         | 0000:41:00.0  | 0           0    / 0          58669/ 65536         |
+===========================+===============+====================================================+
| 6     910B3               | OK            | 97.1        48                0    / 0             |
| 0                         | 0000:82:00.0  | 0           0    / 0          58669/ 65536         |
+===========================+===============+====================================================+
| 7     910B3               | OK            | 94.1        51                0    / 0             |
| 0                         | 0000:42:00.0  | 0           0    / 0          58669/ 65536         |
+===========================+===============+====================================================+
+---------------------------+---------------+----------------------------------------------------+
| NPU     Chip              | Process id    | Process name             | Process memory(MB)      |
+===========================+===============+====================================================+
| 0       0                 | 8769          |                          | 55345                   |
+===========================+===============+====================================================+
| 1       0                 | 8773          |                          | 55345                   |
+===========================+===============+====================================================+
| 2       0                 | 8774          |                          | 55345                   |
+===========================+===============+====================================================+
| 3       0                 | 8775          |                          | 55345                   |
+===========================+===============+====================================================+
| 4       0                 | 8776          |                          | 55345                   |
+===========================+===============+====================================================+
| 5       0                 | 8777          |                          | 55345                   |
+===========================+===============+====================================================+
| 6       0                 | 8778          | python3                  | 55344                   |
+===========================+===============+====================================================+
| 7       0                 | 8779          |                          | 55345                   |
+===========================+===============+====================================================+

CANN:
package_name=Ascend-cann-toolkit
version=9.0.1
innerversion=V100R001C10SPC002B220
compatible_version=[V100R001C21],[V100R001C22],[V100R001C23],[V100R001C25],[V100R001C10]
arch=aarch64
os=linux
path=/usr/local/Ascend/cann-9.0.1
```

</details>

### 🐛 Describe the bug

vllm_ascend/utils.py 中 register_ascend_customop 注册 OOT 算子替换时，SiluAndMulWithClamp 的 key 写成了 "SiluAndMulClamp"，与原始类名 "SiluAndMulWithClamp"不一致，导致 AscendSiluAndMulWithClamp 从未替换生效。

  原因

  vllm 的 CustomOp.__new__ 通过 类名 cls.__name__ 查 op_registry_oot 来决定是否替换为 Ascend 实现：

  # vllm/model_executor/custom_op.py
  class CustomOp(nn.Module):
      def __new__(cls, *args, **kwargs):
          op_name = cls.__name__                     # "SiluAndMulWithClamp"
          if op_name not in op_registry_oot:         # 查 "SiluAndMulWithClamp"
              op_cls_to_instantiate = cls            # 查不到 → 不替换，用原始类
          else:
              op_cls_to_instantiate = op_registry_oot[op_name]
          return super().__new__(op_cls_to_instantiate)

  而 vllm_ascend/utils.py 注册时：

  REGISTERED_ASCEND_OPS = {
      ...
      "SiluAndMulClamp": AscendSiluAndMulWithClamp,   # ← key 应为 "SiluAndMulWithClamp"
      ...
  }
  for name, op_cls in REGISTERED_ASCEND_OPS.items():
      CustomOp.register_oot(_decorated_op_cls=op_cls, name=name)

  "SiluAndMulClamp" ≠ "SiluAndMulWithClamp"，op_registry_oot 里存的 key 是 "SiluAndMulClamp"，但 __new__ 查的是 "SiluAndMulWithClamp"，查不到，实例化时不会替换为
  AscendSiluAndMulWithClamp。

  影响

  SiluAndMulWithClamp（DeepSeek V4 等模型的 MoE 共享专家使用）实例化的始终是 vllm 原始类，而非 AscendSiluAndMulWithClamp。实际走的是 CustomOp 基类默认的 forward_oot →
  forward_native（纯 PyTorch 的 clamp + sigmoid + mul），功能正确但未使用 NPU 融合算子，存在性能损失。该问题不会报错，难以察觉。

  对比

  其他算子（如 SiluAndMul）的 key 与类名一致，替换正常生效：

  "SiluAndMul": AscendSiluAndMul,   # key == cls.__name__ ✓

  修复

  - "SiluAndMulClamp": AscendSiluAndMulWithClamp,
  + "SiluAndMulWithClamp": AscendSiluAndMulWithClamp,

  验证

  - 修复前：AscendSiluAndMulWithClamp.forward_oot 不会被调用，走纯 PyTorch forward_native
  - 修复后：SiluAndMulWithClamp 实例化时被替换为 AscendSiluAndMulWithClamp，forward_oot 正常调用，走 NPU 融合算子

### PR DESCRIPTION
**Pull request by lj361588364**
### What this PR does / why we need it?

Fixes #16500.

`REGISTERED_ASCEND_OPS` in `vllm_ascend/utils.py` registered the key `"SiluAndMulClamp"`, but `CustomOp.__new__` looks up `op_registry_oot` by the vLLM class name `"SiluAndMulWithClamp"`. Since the two never match, `AscendSiluAndMulWithClamp` was never instantiated — models using `SiluAndMulWithClamp` (e.g. DeepSeek V4 shared experts) silently fell back to the pure-PyTorch `forward_native` instead of the NPU fused op. No error is raised, so this is easy to miss; it only shows up as a performance loss.

This PR corrects the key and adds a regression UT that asserts every registered op key matches the vLLM class name it replaces, so this class of typo fails loudly in the future.

### Does this PR introduce _any_ user-facing change?

No behavior change other than the NPU fused op now actually taking effect as intended. No API change.

### How was this patch tested?

- Added `test_registered_ascend_ops_keys_match_vllm_class_names` in `tests/ut/test_utils.py`: with the bug present it fails, after the fix it passes (verified both ways)
- `pytest tests/ut/test_utils.py`: 32 passed, 1 skipped
- Full `pytest tests/ut` on CPU: failure set identical to the clean main baseline (local vllm-nightly version skew, unrelated to this change); no new failures introduced
- `ruff check` / `ruff format --check` (v0.14.0) and pre-commit hooks pass on the changed files

- vLLM main: https://github.com/vllm-project/vllm/commit/84030bbe3d74d99bad477a3d2e37a973ccd8865c

### CODE REVIEW

File: tests/ut/test_utils.py [hunk 1]
```diff
@@ -371,6 +371,23 @@ def test_register_ascend_customop(
...
 
+    def test_registered_ascend_ops_keys_match_vllm_class_names(self):
+        """Regression test for https://github.com/vllm-project/vllm-ascend/issues/16500.
+
+        CustomOp.__new__ looks up op_registry_oot by the in-tree vLLM class
+        name (cls.__name__), so every REGISTERED_ASCEND_OPS key must equal
+        the name of the vLLM class the Ascend op replaces (i.e. one of its
+        base classes). A mismatched key silently disables the NPU fused op.
+        """
+        for name, op_cls in REGISTERED_ASCEND_OPS.items():
```
[the remark this answers is not in the document]
Comment (lj361588364) [reply tests/ut/test_utils.py]: the test now grabs (name, op_cls) straight from the register_oot calls instead of the global, fixed in 981f9e8c2
